File: demo/cpp/Theories.cpp
#include "ExpectedOutcomeAttributes.h"

using namespace System;
using namespace System::Collections::Generic;
using namespace System::Text;
using namespace NUnit::Framework;

namespace NUnitTestDemo
{
    public class Theories
    {
	private:
        [Datapoints]
        int data[3] = { 0, 1, 42 };

	public:
        [Theory, ExpectPass]
        void Theory_AllCasesSucceed(int a, int b)
        {
            Assert::That(a + b, Is::EqualTo(b + a));
        }

        [Theory, ExpectMixed]
        void Theory_SomeCasesAreInconclusive(int a, int b)
        {
            Assume::That(b != 0);
        }

        [Theory, ExpectMixed]
        void Theory_SomeCasesFail(int a, int b)
        {
            Assert::That(b != 0);
        }
	};
}

File: demo/vb/Theories.vb
Imports NUnit.Framework

Namespace NUnitTestDemo

    Public Class Theories

        <Datapoints>
        Private data As Integer() = {0, 1, 42}

        <Theory, ExpectPass>
        Public Sub Theory_AllCasesSucceed(a As Integer, b As Integer)
            Assert.That(a + b, Iz.EqualTo(b + a))
        End Sub

        <Theory, ExpectMixed>
        Public Sub Theory_SomeCasesAreInconclusive(a As Integer, b As Integer)
            Assume.That(b <> 0)
        End Sub

        <Theory, ExpectMixed>
        Public Sub Theory_SomeCasesFail(a As Integer, b As Integer)
            Assert.That(b <> 0)
        End Sub

    End Class

End Namespace
